python/adjoint/basis.py

```diff
@@ -65,26 +65,26 @@ def __init__(self, resolution, symmetry=None, **kwargs):
             self.symmetry = symmetry
 
         if mp.X in set(self.symmetry):
-            self.Nx = int(resolution * self.volume.size.x / 2)
+            self.Nx = int((int(resolution * self.volume.size.x) + 1) / 2)
```

Comment:
Shouldn't we add 1 after dividing by 2, not before?

Reply:
Fixed.

Reply:
Note that `basis.py` isn't used anywhere -- we just kept it for legacy purposes I think.